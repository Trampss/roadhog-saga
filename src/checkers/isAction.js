export default (action) => {
  if (!/.*_.*/.test(action)) {
    throw new Error(`Wrong format for action: '${action}'. should be '<METHOD_NAME>_<RESOURCE_NAME>' (ie: GET_USERS)`)
  }
}

