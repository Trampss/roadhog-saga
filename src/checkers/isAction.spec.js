/* eslint-env jest */
import isAction from './isAction'

describe('isAction', () => {
  it('should throw an error', () => {
    const test = (action) => {
      let error = false
      try {
        isAction(action)
      } catch (ex) {
        error = true
        expect(ex).toMatchSnapshot()
      }
      expect(error).toBe(true)
    }

    test('simple-string')
    test()
    test(null)
  })

  it('should not throw an error', () => {
    isAction('NICE_ACTION')
  })
})
