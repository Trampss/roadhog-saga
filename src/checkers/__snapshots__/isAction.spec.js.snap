exports[`isAction should throw an error 1`] = `[Error: Wrong format for action: 'simple-string'. should be '<METHOD_NAME>_<RESOURCE_NAME>' (ie: GET_USERS)]`;

exports[`isAction should throw an error 2`] = `[Error: Wrong format for action: 'undefined'. should be '<METHOD_NAME>_<RESOURCE_NAME>' (ie: GET_USERS)]`;

exports[`isAction should throw an error 3`] = `[Error: Wrong format for action: 'null'. should be '<METHOD_NAME>_<RESOURCE_NAME>' (ie: GET_USERS)]`;
